src/libsyntax/parse/token.rs

```diff
@@ -602,6 +643,36 @@ impl Token {
     }
 }
 
+impl Token {
+    crate fn new(kind: TokenKind, span: Span) -> Self {
+        Token { kind, span }
+    }
+
+    /// Some token that will be thrown away later.
+    crate fn dummy() -> Self {
+        Token::new(TokenKind::Whitespace, DUMMY_SP)
+    }
+
+    /// Return this token by value and leave a dummy token in its place.
+    crate fn take(&mut self) -> Self {
+        mem::replace(self, Token::dummy())
+    }
+}
+
+impl PartialEq<TokenKind> for Token {
```

Comment:
This impl is technically "wrong" it's not symmetric, and, if we made it symmetric, it would not be transitive. 

I've tried to remove it, but got a dozen of screen-fulls of errors :D So I guess it's better to leave it as is: seems harmless, even if surprising a bit.